solutions/usaco-923.mdx

```diff
@@ -0,0 +1,420 @@
+---
+id: usaco-923
+source: USACO Gold 2019 February
+title: Painting the Barn
+author: Kevin Sheng
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_paintbarn_gold_feb19.html)
+
+## Explanation
+
+### Initial Observations
+
+**Note:** This solution assumes one has already had a firm grasp of the
+[silver](http://www.usaco.org/current/data/sol_paintbarn_silver_feb19.html)
+version of this problem.
+
+We start off by calculating the layers of paint over each cell of the barn with
+prefix sums, but from there it's a bit hard to determine which layers we should
+paint. It's hard to directly calculate it from this array, but notice that there's
+precisely three states a cell can be in:
+
+1. If it's painted over, it will add $1$ to the area of optimal layers of paint.
```

Comment:
combine 1 and 2

Reply:
they're different types?

Reply:
I meant
"if its painted over, either it will add 1 or sub 1"

Reply:
will wait for third opinion

Reply:
> I meant "if its painted over, either it will add 1 or sub 1"

yeah imo this makes more sense